diff -r -C3 boost_asio_0_3_7/boost/asio/detail/socket_ops.hpp boost_asio_patch/boost/asio/detail/socket_ops.hpp
*** boost_asio_0_3_7/boost/asio/detail/socket_ops.hpp	Tue Jun 20 06:44:27 2006
--- boost_asio_patch/boost/asio/detail/socket_ops.hpp	Wed May  2 17:58:02 2007
***************
*** 29,34 ****
--- 29,38 ----
  #include <boost/asio/error.hpp>
  #include <boost/asio/detail/socket_types.hpp>
  
+ #if defined(__FreeBSD__) && !defined(EAI_NODATA)
+ # define EAI_NODATA 7
+ #endif
+ 
  namespace boost {
  namespace asio {
  namespace detail {
diff -r -C3 boost_asio_0_3_7/boost/asio/error.hpp boost_asio_patch/boost/asio/error.hpp
*** boost_asio_0_3_7/boost/asio/error.hpp	Tue Jun 20 06:44:26 2006
--- boost_asio_patch/boost/asio/error.hpp	Wed May  2 17:57:26 2007
***************
*** 289,295 ****
        try
        {
          char buf[256] = "";
!         what_.reset(new std::string(strerror_r(code_, buf, sizeof(buf))));
          return what_->c_str();
        }
        catch (std::exception&)
--- 289,296 ----
        try
        {
          char buf[256] = "";
! 	strerror_r(code_, buf, sizeof(buf));
!         what_.reset(new std::string(buf));
          return what_->c_str();
        }
        catch (std::exception&)
